### ISSUE TITLE
Unicode in Windows workaround

### ISSUE THREAD

**Issue by keddelzz** [morganey-lang/Morganey#196]
Opening a new issue which repeats the workaround proposed in [95-comment](https://github.com/rexim/Morganey/issues/95#issuecomment-247303317).

Bringing all information here:

> As long the issue #95  is not resolved I propose implementing a workaround, which prints `\` instead of `λ` (on windows only). This makes the result of an evaluation copy-pasteable as the input of another evaluation.

**Comment by rexim** [comment]
Accepted! Thanks!

**Comment by keddelzz** [comment]
This might not be a good idea after all: Six unit-tests have to be rewritten in order to succeed, if this workaround gets implemented!

## 

Should we implement this workaround, nevertheless?

**Comment by rexim** [comment]
@keddelzz Yeah, I see what you mean. The situation is even more complicated than I thought... I'm falling asleep at the moment so I cannot say anything adequate right now. I'll think about that tomorrow and then answer here. Let's postpone the workaround until then. Thanks!

**Comment by ForNeVeR** [comment]
I propose to define the tests in question as platform-specific using [tagging](http://www.scalatest.org/user_guide/tagging_your_tests).

**Comment by rexim** [comment]
Alright, basically we have only two places where we have the lambda symbol hardcoded:
- [LambdaTerm.scala:94](https://github.com/rexim/Morganey/blob/db15b3b438ce722167a882a4cecdbfab7162d5de/kernel/src/main/scala/me/rexim/morganey/ast/LambdaTerm.scala#L94)
- [Main.scala:66](https://github.com/rexim/Morganey/blob/db15b3b438ce722167a882a4cecdbfab7162d5de/src/main/scala/me/rexim/morganey/Main.scala#L66)

Please, correct me if I'm wrong.

I assume the tests that require to be rewritten are the pretty printer tests. Can we just make them more flexible. For example,

``` scala
val lambdas = List("\\\\", "λ")

def assertEqual(expectedTemplate: String, actualResult: String): Boolean =
  lambdas
    .map(expectedTemplate.replaceAll("\\?", _))
    .contains(actualResult)

assertEqual("?x.x", m"\\x . x".toString)
```

If so, I'm ok with implementing of all of that.

**Comment by keddelzz** [comment]
Another place where the lambda symbol is hardcoded is [`Language.lambdaLetter`](https://github.com/rexim/Morganey/blob/db15b3b438ce722167a882a4cecdbfab7162d5de/kernel/src/main/scala/me/rexim/morganey/syntax/Language.scala#L25).

**Comment by rexim** [comment]
> Another place where the lambda symbol is hardcoded is [`Language.lambdaLetter`](https://github.com/rexim/Morganey/blob/db15b3b438ce722167a882a4cecdbfab7162d5de/kernel/src/main/scala/me/rexim/morganey/syntax/Language.scala#L25).

Ah, yeah. Thanks for correcting me! I was looking only at lambdas that go to the output. I double checked everything just in case. Looks like we use this one only for parsing.

**Comment by ForNeVeR** [comment]
_(Moved to https://github.com/morganey-lang/Morganey/issues/95#issuecomment-265112850.)_
